### ISSUE TITLE
source file will be deleted by default after importing,How to avoid?

### ISSUE THREAD

**Issue by cg15** [BlackbitDigitalCommerce/pimcore-data-director#198]
My version is 3.7. Now the source file will be deleted by default after importing. I checked the source code and found that the attribute --rm=true is set in vendor/blackbit/data-director/Command/ImportCompleteCommand.php. How can I set it so that the import does not set this attribute?
![image](https://github.com/user-attachments/assets/651fec19-5c3f-4800-869a-8d3952ab7801)
![image](https://github.com/user-attachments/assets/a54994ed-dbfe-406d-b4e8-9a26591042a3)

**Comment by cg15** [comment]
the root cause is ($importFileName ? ' --rm':'') in \Blackbit\DataDirectorBundle\Controller\ImportController::manualImportAction,
**I now want to use the source file and use dd:complete in the callback to do another import, but the source file has been deleted, causing the import to fail**
```
                            $cmd = '"' .Cli::getPhpCli() . '" "'.realpath(PIMCORE_PROJECT_ROOT . DIRECTORY_SEPARATOR . 'bin' . DIRECTORY_SEPARATOR . 'console') . '" data-director:complete ' . $dataportId . ' "'.$importFileName.'"'.($importFileName ? ' --rm':'').' -n --status-key='.$statusKey.' -f'.(($locale !== null) ? ' --locale='.$locale : '').($request->get('dry-run') ? ' --dry-run':'').($parameters ? ' --parameters='.escapeshellarg(json_encode($parameters)):'').' --user='.Tool\Admin::getCurrentUser()->getId();
```

**Comment by BlackbitDevs** [comment]
Hi @cg15,

tl;dr
In 3.7.68 I fixed this trap by not deleting the manually uploaded file immediately after the run. This way the follow-up dataport can still use it. It will later get deleted by the maintenance job.

Longer response:
There are multiple ways to solve this:
1. Add a raw data field with selector `__source`. This will contain the path to a temporary copy of your source file, see https://blackbitdigitalcommerce.github.io/pimcore-data-director/#dataport-settings:
   > In addition to the source file fields you can use the field __source to obtain the source file name. This can be useful if you want to process the source file name in the object import (e.g. to call a dependent import with the same import file). It will contain a local file copy of the import resource to ensure that even if you use the --rm option the file will be available to follow-up imports.
2. The follow-up dataport could reuse the raw data from first dataport. For this you can right click the first dataport and select "Create raw data report" (from Pimcore 11 the `PimcoreCustomReportsBundle` needs to be installed for this to work). And then change the second dataport to type "Pimcore report", select the just created report as source and add the columns which you need for the follow-up dataport as raw data.
With this approach it is not necessary to read the import file again.
3. In 3.7.68 I fixed this trap by not deleting the manually uploaded file immediately after the run. This way the follow-up dataport can still use it. It will later get deleted by the maintenance job.

**Comment by cg15** [comment]
@BlackbitDevs 
thank you very much
